### PR TITLE
Add CI run of nox on GitHub Actions

### PR DESCRIPTION
**Pull request by adamchainz**
Ensure all tests are passing for every PR and commit.

This is based on the standard config I use for my projects, but adapted for nox.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+jobs:
+  tests:
+    name: Python ${{ matrix.python-version }}
+    runs-on: ubuntu-20.04
```
Comment (JaapJoris) [inline .github/workflows/main.yml:12]: Why not `ubuntu-latest`?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    strategy:
```
Comment (JaapJoris) [inline .github/workflows/main.yml:14]: Should we add `fail-fast: false`?

**Review by JaapJoris — COMMENTED** [review]
Two questions:

File: .github/workflows/main.yml (hunk 1 above)
Comment (adamchainz) [reply .github/workflows/main.yml:12]: I like to pin things.

File: .github/workflows/main.yml (hunk 2 above)
Comment (adamchainz) [reply .github/workflows/main.yml:14]: I don't have a problem with the default fail fast. You do you though!

**Pull request merged**